InteractionBase/src/vis/VViewSwitcherEntry.cpp

```diff
@@ -0,0 +1,59 @@
+/***********************************************************************************************************************
+ **
+ ** Copyright (c) 2015 ETH Zurich
+ ** All rights reserved.
+ **
+ ** Redistribution and use in source and binary forms, with or without modification, are permitted provided that the
+ ** following conditions are met:
+ **
+ **    * Redistributions of source code must retain the above copyright notice, this list of conditions and the
+ **      following disclaimer.
+ **    * Redistributions in binary form must reproduce the above copyright notice, this list of conditions and the
+ **      following disclaimer in the documentation and/or other materials provided with the distribution.
+ **    * Neither the name of the ETH Zurich nor the names of its contributors may be used to endorse or promote products
+ **      derived from this software without specific prior written permission.
+ **
+ **
+ ** THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES,
+ ** INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ ** DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ ** SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ ** SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+ ** WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ ** OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ **
+ **********************************************************************************************************************/
+#include "VViewSwitcherEntry.h"
+
+#include "VisualizationBase/src/declarative/DeclarativeItemDef.h"
+#include "vis/TextAndDescription.h"
+#include "VisualizationBase/src/items/Text.h"
+#include "VisualizationBase/src/items/ViewItem.h"
+
+#include "VisualizationBase/src/cursor/TextCursor.h"
+
+namespace Interaction {
+
+ITEM_COMMON_DEFINITIONS(VViewSwitcherEntry, "item")
+
+VViewSwitcherEntry::VViewSwitcherEntry(Visualization::Item* parent, NodeType* node, const StyleType* style) :
+		Super(parent, node, style)
+{
+	nameField_ = new Visualization::Text(this, node->viewName());
```

Comment:
Looking at this, you can simplify this class quite a bit and save both of these lines here, by using VText as opposed to Text for the type of `nameField_`. VText requires that you provide it with a node, so you can directly use the `node->viewName()` in `initializeForms`. Then the text + editing + node renaming will happen automatically.

Please do that, as it will make the class smaller, and it's the right thing to have here.